### PR TITLE
`Printf`: document format magic numbers

### LINKED ISSUE #41649: Document magic numbers in Printf for Floats
For Ints and Chars (in the code above the one I link) it's somewhat more straightforward what the numbers mean and why they're chosen, but for Floats it's kind of guesswork unless you already know what they mean. This came up while investigating #41631

https://github.com/JuliaLang/julia/blob/6279f3b8c4a7e7a669cc7222642ae20824292855/stdlib/Printf/src/Printf.jl#L779-L782

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/JuliaLang/julia/issues/41649.

### CODE REVIEW

File: stdlib/Printf/src/Printf.jl [hunk 1]
```diff
@@ -626,11 +626,22 @@ end
 function fix_dec end
 function ini_dec end
 
+"maximal exponent for `Float64`s: `ceil(Int, log10(prevfloat(typemax(Float64))))`"
+max_integer_part_width() = 309
```
Comment (KristofferC) [inline stdlib/Printf/src/Printf.jl:630]: Why are these functions?

Comment (t-bltg) [reply stdlib/Printf/src/Printf.jl:630]: > Why are these functions?

It allows to reuse magic numbers in multiple places in this file. Should they be `const` globals instead ?

> Also, seems like the string should just be a comment?

Yes, we could make them a comment, but it's imo better to leave them as a docstring.

Comment (KristofferC) [reply stdlib/Printf/src/Printf.jl:630]: > Should they be uppercased const globals instead ?

Yes, it seems like just consts is more lightweight than having to define a function for every single number.

> Any argument against?

The docstring has to be stored somewhere etc. This just seems like a completely standard scenario where you would use a comment.

Comment (t-bltg) [reply stdlib/Printf/src/Printf.jl]: Allright, suggestions applied.

**Comment by t-bltg** [comment]
CI errors unrelated: bump, merge ?

**Review by quinnj — APPROVED** [review]

**Pull request merged**